### ISSUE TITLE
Support archiving the files (zip)

### ISSUE THREAD

**Issue by nishantkagrawal** [gregnb/filemanager-webpack-plugin#20]
## Request

archiving the files for deployment is common for us. I have been using this plugin and found it helpful to setup the right deployment directory. I would also like to use this as an archive tool.

Is it possible?

**Comment by gregnb** [comment]
It sounds interesting, but let me think about this some more.  So on every build you guys archive it for historical purposes?

I don't want to throw just every feature in the book and bloat this package, but I'm open minded to maybe the possibility of people needing to archive builds.  Just curious as to the frequency in which you do it?

**Comment by nishantkagrawal** [comment]
We actually archive it and then the deployment servers would extract them and deploy. So yes each and every build with create the archive.

**Comment by gregnb** [comment]
This feature has been added and available in 1.0.22

**Comment by nishantkagrawal** [comment]
Excellent! Thanks for the quick update!

Is it possible to update the library to take the archiver parameters to be passed from within the configuration? 

So basically I would like to 

```
onEnd: {               
	archive: [{
		source : path.resolve(baseDir, '.tmp/dist'), 
		destination: path.resolve(baseDir, 'deploy/SampleApplication.tar'), 
		archiver: {
			format: 'tar',
			options: {
				gzip: true,
				gzipOptions: {
					level: 1
				}
			}
		}
	}]
}
```

So the format will be the first parameter of the archiver function (with zip as default) and the options would be the second parameter.

https://github.com/archiverjs/node-archiver/tree/master/examples

**Comment by gregnb** [comment]
For now, it will have to stay the way it is. My reasoning for it is i plan on adding in a "global" options parameter so people can tweak not only this but the cpx options for copying and other things.

**Comment by nishantkagrawal** [comment]
But in archive case, wouldnt it be necessary to have options scoped to the archive object, as there could be multiple archive outputs with different formats?

**Comment by gregnb** [comment]
Alright I'll look into it tonight. Thanks

**Comment by gregnb** [comment]
Here's what I'm thinking instead of having 'archive' repeat twice. That seems confusing to me:

```
        archive: [
          { 
            source: "./dist", 
            destination: "./testing/test1.zip" 
            format: 'zip' or 'tar',
            options: {
              ...whateverOptionsForArchive..
            }
          },
```

If no format or options are provided it defaults to zip

**Comment by nishantkagrawal** [comment]
Works for me! Thanks!

**Comment by gregnb** [comment]
This support is now available from 1.0.23

**Comment by nishantkagrawal** [comment]
Perfect!
